World/Environment/Lightning.tscn

```diff
@@ -1,56 +1,88 @@
-[gd_scene load_steps=5 format=3 uid="uid://cib4g5bstb1xa"]
+[gd_scene load_steps=10 format=3 uid="uid://cib4g5bstb1xa"]
 
 [ext_resource type="Script" path="res://World/Environment/Lightning.gd" id="1_2eur0"]
+[ext_resource type="Texture2D" uid="uid://b3xdbcxmrvtvn" path="res://World/Environment/lightning_cloud_lighting.png" id="2_yil0v"]
+
+[sub_resource type="Gradient" id="Gradient_gnng6"]
+offsets = PackedFloat32Array(0, 0.431373, 1)
+colors = PackedColorArray(0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 0)
+
+[sub_resource type="Gradient" id="Gradient_kbe6e"]
+offsets = PackedFloat32Array(1)
+colors = PackedColorArray(1, 1, 1, 0)
 
 [sub_resource type="ViewportTexture" id="ViewportTexture_schoq"]
-viewport_path = NodePath("MeshInstance3d/SubViewport")
+viewport_path = NodePath("LightningMesh/SubViewport")
 
 [sub_resource type="StandardMaterial3D" id="StandardMaterial3D_dju3a"]
 resource_local_to_scene = true
 transparency = 1
+cull_mode = 2
 albedo_texture = SubResource("ViewportTexture_schoq")
 emission_enabled = true
 emission = Color(0.462745, 0.537255, 1, 1)
-emission_energy_multiplier = 0.8
+emission_energy_multiplier = 10.0
 billboard_mode = 2
 
 [sub_resource type="PlaneMesh" id="PlaneMesh_h6sm4"]
-size = Vector2(150, 750)
+size = Vector2(500, 500)
 orientation = 2
 
+[sub_resource type="StandardMaterial3D" id="StandardMaterial3D_x464h"]
+resource_local_to_scene = true
+transparency = 1
+cull_mode = 2
+albedo_color = Color(1, 1, 1, 0)
+albedo_texture = ExtResource("2_yil0v")
+emission_enabled = true
+emission = Color(0.462745, 0.537255, 1, 1)
+emission_energy_multiplier = 10.0
+billboard_mode = 2
+
+[sub_resource type="QuadMesh" id="QuadMesh_jseao"]
+size = Vector2(200, 200)
+
 [node name="Lightning" type="Node3D"]
 script = ExtResource("1_2eur0")
-eye_path = NodePath("Camera3D")
+enabled = true
+spawn_interval_to = 3.0
+gradient_on = SubResource("Gradient_gnng6")
+gradient_off = SubResource("Gradient_kbe6e")
+color = Color(0.788235, 0.776471, 1, 1)
+rot_degrees = 180.0
+
+[node name="Timer" type="Timer" parent="."]
+autostart = true
 
-[node name="MeshInstance3d" type="MeshInstance3D" parent="."]
-transform = Transform3D(-4.37114e-08, 0, -1, 0, 1, 0, 1, 0, -4.37114e-08, -443.156, -88.0528, -0.99136)
+[node name="LightningMesh" type="MeshInstance3D" parent="."]
 material_override = SubResource("StandardMaterial3D_dju3a")
 mesh = SubResource("PlaneMesh_h6sm4")
 
-[node name="SubViewport" type="SubViewport" parent="MeshInstance3d"]
+[node name="LitCloudMesh" type="MeshInstance3D" parent="LightningMesh"]
+transform = Transform3D(1, 0, 0, 0, 1, 0, 0, 0, 1, 0, 234.387, 0)
+material_override = SubResource("StandardMaterial3D_x464h")
+mesh = SubResource("QuadMesh_jseao")
+
+[node name="SubViewport" type="SubViewport" parent="LightningMesh"]
 disable_3d = true
 transparent_bg = true
 handle_input_locally = false
 msaa_2d = 3
 use_taa = true
-size = Vector2i(100, 500)
+size = Vector2i(500, 500)
 render_target_update_mode = 4
```

Comment:
the performance hit is probably really small but since SubViewports are not too well optimized to begin with it might be worthwhile to be careful: I think it'd be better to set this to `UPDATE_ONCE` right after generating a lighting line rather than re-rendering the line to this viewport every frame.

Reply:
I like that you think of small details :D
However we need to render more than once for the flickering animation. I still tried to increase performance by setting the render_mode to `VISIBLE_PARENT` and manually set the visibility only once the animation starts.

[632dd3b](https://github.com/boku-ilen/landscapelab/pull/326/commits/632dd3b50e4ea0b8e6a674f468e76c416aeed171)

Reply:
Ah that makes sense! sounds good :+1: